tools/publisher/src/subcommand/fix_manifests/validate.rs

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+
+use crate::fs::Fs;
+use crate::package::discover_and_validate_package_batches;
+use anyhow::{anyhow, bail, Result};
+use semver::Version;
+use std::collections::BTreeMap;
+use tracing::info;
+
+/// Validations that run before the manifests are fixed.
+///
+/// For now, this validates:
+/// - `aws-config` version number matches all `aws-sdk-` prefixed versions
+/// - `aws-smithy-` prefixed versions match `aws-` (NOT `aws-sdk-`) prefixed versions
+pub(super) fn pre_validate_manifests(versions: &BTreeMap<String, Version>) -> Result<()> {
+    info!("Pre-validation manifests...");
+    let expected_sdk_version = versions
+        .get("aws-config")
+        .ok_or_else(|| anyhow!("`aws-config` crate missing"))?;
+    let expected_smithy_version = versions
+        .get("aws-smithy-types")
+        .ok_or_else(|| anyhow!("`aws-smithy-types` crate missing"))?;
+
+    for (name, version) in versions {
+        if name.starts_with("aws-smithy-") {
+            confirm_version(name, expected_smithy_version, version)?;
```

Comment:
I know we had some crate classification logic somewhere already, should we share that here?

Reply:
Ah, yeah. Forgot about that. Will fix.